redis/redisLimiter.go

```diff
@@ -195,6 +196,72 @@ func (rl *rateLimiter) getFailConfig(mode Mode) (time.Duration, int64) {
 	return rl.freezeFailureConfig.limitPeriod, rl.freezeFailureConfig.maxFailures
 }
 
+func (rl *rateLimiter) getSecurityFailedTrials(ctx context.Context, key string) (uint64, error) {
+	failedTrials, err := rl.storer.GetValue(ctx, key)
+	if err != nil {
+		if err.Error() == "redis: nil" {
+			return 0, ErrKeyNotExists
+		}
+		return 0, err
+	}
+	convertedValue, err := strconv.ParseUint(failedTrials, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return convertedValue, nil
+}
+
+// SetSecurityMode will set the key from volatile to persistent
+func (rl *rateLimiter) SetSecurityMode(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	wasSet, err := rl.storer.SetPersist(ctx, key)
+
+	if !wasSet {
```

Comment:
I think this can be removed as ErrKeyNotExists might not be correct here, maybe add a log here:
```go
if !wasSet {
	log.Warn("security mode was not set")
}
```